diff --git a/sites/all/modules/views/js/ajax.js b/sites/all/modules/views/js/ajax.js
index 6669c85..98787fe 100644
--- a/sites/all/modules/views/js/ajax.js
+++ b/sites/all/modules/views/js/ajax.js
@@ -54,8 +54,8 @@ Drupal.Views.Ajax.ajaxResponse = function(data) {
       var ajax_title = Drupal.settings.views.ajax.title;
 
       // Bind a click to the button to set the value for the button.
-      $('input[type=submit], button', ajax_area).unbind('click');
-      $('input[type=submit], button', ajax_area).click(function() {
+      $('input[type="submit"], button', ajax_area).unbind('click');
+      $('input[type="submit"], button', ajax_area).click(function() {
         $('form', ajax_area).append('<input type="hidden" name="'
           + $(this).attr('name') + '" value="' + $(this).val() + '">');
         $(this).after('<span class="views-throbbing">&nbsp</span>');
@@ -169,8 +169,8 @@ Drupal.Views.Ajax.previewResponse = function(data) {
     // if a URL was supplied, bind the form to it.
     if (url) {
       // Bind a click to the button to set the value for the button.
-      $('input[type=submit], button', ajax_area).unbind('click');
-      $('input[type=submit], button', ajax_area).click(function() {
+      $('input[type="submit"], button', ajax_area).unbind('click');
+      $('input[type="submit"], button', ajax_area).click(function() {
         $('form', ajax_area).append('<input type="hidden" name="'
           + $(this).attr('name') + '" value="' + $(this).val() + '">');
         $(this).after('<span class="views-throbbing">&nbsp</span>');
@@ -199,7 +199,7 @@ Drupal.Views.updatePreviewForm = function() {
   var url = $(this).attr('action');
   url = url.replace('nojs', 'ajax');
 
-  $('input[type=submit], button', this).after('<span class="views-throbbing">&nbsp</span>');
+  $('input[type="submit"], button', this).after('<span class="views-throbbing">&nbsp</span>');
   $(this).ajaxSubmit({
     url: url,
     data: { 'js': 1 },
@@ -216,8 +216,8 @@ Drupal.Views.updatePreviewFilterForm = function() {
   var url = $(this).attr('action');
   url = url.replace('nojs', 'ajax');
 
-  $('input[type=submit], button', this).after('<span class="views-throbbing">&nbsp</span>');
-  $('input[name=q]', this).remove(); // remove 'q' for live preview.
+  $('input[type="submit"], button', this).after('<span class="views-throbbing">&nbsp</span>');
+  $('input[name="q"]', this).remove(); // remove 'q' for live preview.
   $(this).ajaxSubmit({
     url: url,
     data: { 'js': 1 },
diff --git a/sites/all/modules/views/js/ajax_view.js b/sites/all/modules/views/js/ajax_view.js
index e56e73a..3f29e05 100644
--- a/sites/all/modules/views/js/ajax_view.js
+++ b/sites/all/modules/views/js/ajax_view.js
@@ -64,7 +64,7 @@ Drupal.behaviors.ViewsAjaxView = function() {
         // remove 'q' from the form; it's there for clean URLs
         // so that it submits to the right place with regular submit
         // but this method is submitting elsewhere.
-        $('input[name=q]', this).remove();
+        $('input[name="q"]', this).remove();
         var form = this;
         // ajaxSubmit doesn't accept a data argument, so we have to
         // pass additional fields this way.
@@ -74,7 +74,7 @@ Drupal.behaviors.ViewsAjaxView = function() {
       })
       .addClass('views-processed')
       .submit(function () {
-        $('input[type=submit], button', this).after('<span class="views-throbbing">&nbsp</span>');
+        $('input[type="submit"], button', this).after('<span class="views-throbbing">&nbsp</span>');
         var object = this;
         $(this).ajaxSubmit({
           url: ajax_path,
